docs/overview.md

```diff
@@ -0,0 +1,385 @@
+# Overview
+
+This is an overview of most language features in Reason. It does not explain
+them in detail, but should serve as a quick reference. Please see the guides
+on the left for additional details about each feature.
+
+## Let Bindings
+
+_Details: [Let Bindings](/let-bindings)_
+
+Feature                         | Example
+--------------------------------|----------
+String value                    | <code class="text-ocaml">let hi = "Hello World"</code><code class="text-reasonml">let hi = "Hello World";</code>
+Int value                       | <code class="text-ocaml">let count = 42</code><code class="text-reasonml">let count = 42;</code>
+Type annotation on binding      | <code class="text-ocaml">let count: int = 42</code><code class="text-reasonml">let count: int = 42;</code>
+
+- Note: Let bindings are immutable and cannot change once created.
+
+## Built In Types
+
+_Details: [Primitives](/primitives)_
+
+Feature                         | Example
+--------------------------------|----------
+Int                             | <code class="text-ocaml">let x: int = 10</code><code class="text-reasonml">let x: int = 10;</code>
+Float                           | <code class="text-ocaml">let x: float = 10.0</code><code class="text-reasonml">let x: float = 10.0;</code>
+Boolean                         | <code class="text-ocaml">let x: bool = false</code><code class="text-reasonml">let x: bool = false;</code>
+String                          | <code class="text-ocaml">let x: string = "ten"</code><code class="text-reasonml">let x: string = "ten";</code>
+Char                            | <code class="text-ocaml">let x: char = 'c'</code><code class="text-reasonml">let x: char = 'c';</code>
+Unit                            | <code class="text-ocaml">let x: unit = ()</code><code class="text-reasonml">let x: unit = ();</code>
+Option                          | <code class="text-ocaml">let x: int option = Some 10</code><code class="text-reasonml">let x: option(int) = Some(10);</code>
+Tuple                           | <code class="text-ocaml">let x: (int, string) = (10, "ten")</code><code class="text-reasonml">let x: (int, string) = (10, "ten");</code>
+List                            | <code class="text-ocaml">let x: int list = [1; 2; 3];</code><code class="text-reasonml">let x: list(int) = [1, 2, 3];</code>
+Array                           | <code class="text-ocaml">let x: int array = [&#124;1; 2; 3&#124;]</code><code class="text-reasonml">let x: array(int) = [&#124;1, 2, 3&#124;];</code>
+Functions                       | <code class="text-ocaml">let x : int -> int -> int = fun a b -> a + b</code><code class="text-reasonml">let x: (int, int) => int = (a, b) => a + b;</code>
```

Comment:
```ocaml
let x : int -> int -> int = fun a b -> a + b
``` 

is not the most idiomatic way to define a function in OCaml syntax. But 

```ocaml
let x a b = a + b
```

Doesn't include the type annotation. Or maybe it could be this?

```ocaml
let x (a : int) (b : int) : int = a + b
```